VoiceController.php

```diff
@@ -1,48 +1,30 @@
-public function voice(Request $request){
-    $request->validate([
-        'question_id'=>'required|int|exists:questions,id',
-        'value'=>'required|boolean',
-    ]);
 
+
+public function voice(StoreQuestionRequest $request){
     $question=Question::find($request->post('question_id'));
-    if (!$question)
-        return response()->json([
-            'status'=>404,
-            'message'=>'not found question ..'
-        ]);
-    if ($question->user_id==auth()->id())
-        return response()->json([
-            'status' => 500,
-            'message' => 'The user is not allowed to vote to your question'
+    if (!$question->user_id==auth()->id()){
```

Comment:
@javiershaka haven't launched the code, but pretty sure that `!` in the beginning of the variable means "variable is empty", so it may have incorrect result. Should be `if ($question->user_id != auth()->id())`, and then I'm not sure what happens if the question is not found - it automatically goes down to the final return with 406? May be, but would need to test.

Reply:
Until this point, due to the validation of the StoreQuestionRequest, it must have already been validated, it only sends a search in the database, the condition validates if it is the same person who made the question, therefore if it is true it happens to deny, which is why it throws The 406 exception denied, personally I prefer to use the JsonResponse :: HTTP_UNPROCESSABLE_ENTITY class, but it's a matter of taste.